jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConfigFilesFacade.kt

```diff
@@ -188,6 +194,49 @@ class DefaultConfigFilesFacade(
         }
     }
 
+    override fun deleteSsoConnectionFromConfig(sessionName: String) {
+        val filePath = configPath
+        val lines = filePath.inputStreamIfExists()?.reader()?.readLines().orEmpty()
+        val ssoHeaderLine = lines.indexOfFirst { it.startsWith("[${SsoSessionConstants.SSO_SESSION_SECTION_NAME} $sessionName]") }
+        if (ssoHeaderLine == -1) return
+        val nextHeaderLine = lines.subList(ssoHeaderLine + 1, lines.size).indexOfFirst { it.startsWith("[") }
+        val endIndex = if (nextHeaderLine == -1) lines.size else ssoHeaderLine + nextHeaderLine + 1
+        val updatedArray = lines.subList(0, ssoHeaderLine) + lines.subList(endIndex, lines.size)
+        val profileHeaderLine = getCorrespondingSsoSessionProfilePosition(updatedArray, sessionName)
+        if (profileHeaderLine == -1) {
+            filePath.writeText(updatedArray.joinToString("\n"))
+        } else {
+            val nextHeaderLine2 = updatedArray.subList(profileHeaderLine + 1, updatedArray.size).indexOfFirst { it.startsWith("[") }
+            val endIndex2 = if (nextHeaderLine2 == -1) updatedArray.size else profileHeaderLine + nextHeaderLine2 + 1
+            filePath.writeText((updatedArray.subList(0, profileHeaderLine) + updatedArray.subList(endIndex2, updatedArray.size)).joinToString("\n"))
+        }
+        val applicationManager = ApplicationManager.getApplication()
+        if (applicationManager != null && !applicationManager.isUnitTestMode) {
+            FileDocumentManager.getInstance().saveAllDocuments()
+            ProfileWatcher.getInstance().forceRefresh()
+        }
+    }
+
+    private fun getCorrespondingSsoSessionProfilePosition(updatedArray: List<String>, sessionName: String): Int {
+        var content = updatedArray
+        var lineCnt = 0
+        while (content.size > 0) {
+            val pos = content.indexOfFirst { it.startsWith("[profile $sessionName-") }
+            // if no matching profile section found
+            if (pos == -1) return -1
+
+            // if matching profile section found which is an sso-profile
+            if (isProfileSso(content.subList(pos + 1, content.size))) return lineCnt + pos
+
+            content = content.subList(pos + 1, content.size)
+            lineCnt += pos + 1
+        }
+        return -1
+    }
+    private fun isProfileSso(configContent: List<String>): Boolean {
+        val nextNonCommentedLine = configContent.indexOfFirst { !it.startsWith("#") }
```

Comment:
not sure why we need a comment check

Reply:
what if the `sso_session` is not the first line?

Reply:
Checking the content of the entire profile section to see if we get this key